Lib/sqlite3/dump.py

```diff
@@ -73,9 +78,14 @@ def _iterdump(connection):
                 "||quote({0})||".format(_quote_name(col)) for col in column_names
             )
         )
-        query_res = cu.execute(q)
-        for row in query_res:
-            yield("{0};".format(row[0]))
+        orig_text_factory = connection.text_factory
+        try:
+            connection.text_factory = bytes
+            query_res = cu.execute(q)
+            for row in query_res:
+                yield("{0};".format(_force_decode(row[0])))
+        finally:
+            connection.text_factory = orig_text_factory
```

Comment:
Could you separate the text factory swapping into a context manager helper? Even though it will only be used once, it will make the code more readable.